### ISSUE TITLE
TitleIndicator Rewrite

### ISSUE THREAD

**Issue by JakeWharton** [JakeWharton/ViewPagerIndicator#34]
Rewrite title indicator for better usability and looks.

The control is entirely drawn on a canvas which severely limits its ability to provide accessible use as well as do cool things like provide touch state indication with the adjacent titles.

This should also come with the very nice fading edges that the Market and G+ use... I'm not against baksmali-ing their APKs...

This can probably be done progressively:
- Rewrite movement to start instantly when the paging starts and end precisely when the page has completed scrolling.
- Render current style with `TextView`s
- Add touch-state indication
- Enable edge fading

**Comment by JakeWharton** [comment]
`android.support.v4.view.PagerTitleStrip` would be a good starting point...

**Comment by tprochazka** [comment]
I tested PagerTitleStrip now and it doesn't work better like yours current TitleIndicator :-(
It work almost the same, but has no too much great features.
Indicator in G+ is much better, only one weakness is that it doesn't support click to left|right title to fast swap.

**Comment by tprochazka** [comment]
https://github.com/ChainsDD/Superuser/blob/master/src/com/noshufou/android/su/widget/PagerHeader.java
This is very nice implementation which work better like in G+ or market ;-)

**Comment by JakeWharton** [comment]
I'll happily accept pull requests if someone wants to port that over :)

I'm deep in [ActionBarSherlock](http://github.com/JakeWharton/ActionBarSherlock) development for the next few weeks before I can even think about looking back at this library.

**Comment by tprochazka** [comment]
I just started using https://github.com/astuetz/ViewPagerExtensions - SwipeyTab. Very nice implementation, it has several small bugs, it is more hard to use and setOnCenterClickListener() is missing but looks as I need. It's should be great inspiration...

**Comment by JakeWharton** [comment]
Yeah I've used it. And yes, it has a few bugs. I'm going to provide a generic ViewGroup indicator from which it will be trivial to create title, icon, or custom indicators that use actual views.

**Comment by JakeWharton** [comment]
- Support icons via compound drawables (like with tabs). See #115.
